### PR TITLE
Add preinstall script

### PR DESCRIPTION
**Pull request by ghost**
I haven't tested this because uninstalling wget uninstalls google-chrome for some reason.

### CODE REVIEW

**Comment by ghost** [comment]
@gabru-md Could you test this.

**Comment by gabru-md** [comment]
preinstall.js ? is it missing ?

**Comment by ghost** [comment]
I've added the file. See 6ff481b. I don't know how it never got commited. I usually run `git status` all of the time, so I should have seen it.

File: preinstall.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+var exec = require('child_process').exec;
+var spawn = require('child_process').spawn;
+var readline = require('readline');
+var terminal = require('terminal-utilities')
+
```
Comment (gabru-md) [inline preinstall.js]: chalk is missing from the modules.

File: preinstall.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+var rl = readline.createInterface({
+	input: process.stdin,
+	output: process.stdout
+});
+
+exec('which wget', function(err,stdout,stderr){
+	if(err){
```
Comment (gabru-md) [inline preinstall.js]: there is a problem with the use of if/else here all throughout the code.
if an error is encountered it means that `wget` is not present, and hence must be installed. But the code seems to do something else.

File: preinstall.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		console.log(chalk.redBright('An unknown error occurred'));
+		process.exit(1)
+	}
+	if(stdout){
+		console.log('wget is required to install this module');
+		exec('which apt-get', function(err,stdout,stderr){
+			if(stdout){
```
Comment (gabru-md) [inline preinstall.js]: same goes for here

File: preinstall.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+					if (answer.match(/^y/i)){
+						var installation = spawn('sudo', ['apt-get', 'install', 'wget', '-y'])
+						installation.stdout.on('data', function(data){
+							terminal.write(data);
+						});
+						installation.stderr.on('data', function(data){
+ 							terminal.write(data);
+						});
+						installation.on('close', function(code){
+							console.log('wget installed successfully');
```
Comment (gabru-md) [inline preinstall.js]: process.exit(0) is missing.
the process stops thereafter and has to be manually terminated.

**Review by gabru-md — CHANGES_REQUESTED** [review]
Please do the following changes before merge can be done!

File: preinstall.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+						});
+					}
+				});
+			}
+		}
```
Comment (gabru-md) [inline preinstall.js]: some brackets are missing. fix them as well.
:D

File: preinstall.js (hunk 1 above)
Comment (ghost) [reply preinstall.js]: Fixed. See c74eae2

File: preinstall.js (hunk 2 above)
Comment (yamboy1) [inline preinstall.js]: Fixed. See 56c2571

Comment (ghost) [reply preinstall.js]: Fixed. See 6edb0c1

File: preinstall.js (hunk 3 above)
Comment (ghost) [reply preinstall.js]: Fixed. See 6edb0c1

File: preinstall.js (hunk 4 above)
Comment (ghost) [reply preinstall.js]: Fixed. See 6e6447b

**Comment by ghost** [comment]
@gabru-md I've made the changes that you have requested.

**Comment by gabru-md** [comment]
there were still some issues to be resolved.
I did it myself. changing the `preinstall` to `postinstall` and removing ``process.exit(0)`` at the end. 

due to asynchronous nature of Node.JS, the `exec` command was not executed but the process ended faster. So had to remove it XD

Thanks for the contribution though.
Cheers!

**Review by gabru-md — APPROVED** [review]

**Comment by ghost** [comment]
@gabru-md The reason I set it to be `preinstall` was because I had assumed that if the installation of `wget` failed (e.g. no internet connection) the installation of LinuxSpells would still continue and break when it runs `accio`.

**Pull request merged**